salt/_modules/metalk8s_volumes.py

```diff
@@ -382,20 +385,17 @@ def _open_fd(*args, **kwargs):
 #     returns "devtmpfs       devtmpfs   1932084     0   1932084   0% /dev"
 #
 # So yeah, let's not rely on this…
-def _get_from_blkid(path, key):
-    # Can't use `disk.blkid` because it makes no distinction between the
-    # different return codes…
-    res = __salt__['cmd.run_all']('blkid -p {}'.format(path))
-    retcode = res.get('retcode', 0)
-    if retcode == 0:
-        pat = r'{}="(?P<value>.+)"'.format(key)
-        match = re.search(pat, res['stdout'])
-        return match.groupdict().get('value') if match else None
-    if retcode == -2: # Nothing was detected (device not formatted).
-        return None
-    raise CommandExecutionError(
-        'error while trying to run `{0}`: {1}' .format(cmd, res['stderr'])
-    )
+def _get_from_blkid(path):
+    probe_config = {
+        'superblocks': True,
+        'superblocks_flags': [
+            blkid.SuperblockFlags.UUID, blkid.SuperblockFlags.TYPE
+        ],
+        'partitions': True,
+    }
+    with blkid.Probe.from_filename(path) as probe:
+        probe.configure(probe_config)
```

Comment:
Would it make sense to allow configuration from the 'creation' function directly?

Reply:
Yup, I'll move it there.

Done.